src/services/parser/parsing-utilities.js

```diff
@@ -0,0 +1,56 @@
+import {XmlParser} from "./xml-parser/xml-parser";
+import { validateFieldsHaveSuccessfullyParsed } from "./parsing-validation";
+
+export const parseInteractionId = async message => {
+  const ebxmlAsJson = await new XmlParser().parse(JSON.parse(message).ebXML);
+
+  const parsedFields = {
+    interactionId: ebxmlAsJson['data']['Envelope']['Header']['MessageHeader']['Action']
```

Comment:
currently in this way of drilling down xml, if one of the properties layer between is missing, it will just throw a `Uncaught TypeError: Cannot read properties of undefined` right away, and will not arrive at line 11 validateFieldsHaveSuccessfullyParsed(parsedFields);
maybe use option chain ?. to ensure that validateFieldsHaveSuccessfullyParsed got called?
```suggestion
    interactionId: ebxmlAsJson?.['data']?.['Envelope']?.['Header']?.['MessageHeader']?.['Action']
```

Reply:
👏 Didn't know about this, learned something new!